### DISCUSSION TITLE
JSON responses for challenges

### DISCUSSION THREAD

**Discussion by disbeliefff** [go-acme/lego#2382] [Q&A] [answered]
I’m using the lego library for an internal API and want to return a JSON response when performing DNS-01/HTTP-01 validation. The client sends a challenge, and I expect a JSON response containing the generated value, status, and other relevant details. I’m having trouble implementing this approach. Is it possible with the lego library?

**Comment by ldez** [comment]
Your question is too vague to be answered.

**Reply by disbeliefff** [reply]
I'm trying to write an application that uses lego as a library. I need full control over the generated value of http-01/dns-01 challanges, and certificates to store locally in my database. 

I need when querying for example dns01 to return a json response with the LE generated TXT and order status. However, I am facing a problem that I can return the authz link where this data is stored, but I can't return it as json. Is this possible?

**Reply by ldez** [reply]
why do you need to store the temporary TXT and the order?

**Reply by disbeliefff** [reply]
Since the user makes a request to my service, which makes a request to lego with the domain and type of challenge specified by the user, I need to store this information in my database

**Reply by ldez** [reply]
You don't need to store temporary TXT and order to have a delegation.
Your architectural design seems wrong.

Once again this is too vague to really discuss your implementation.

**Reply by disbeliefff** [reply]
If I understand correctly, you suggest not storing temporary data (such as TXT records or orders) in the database. However, in my case, I need to obtain these values and return them to the user. For example, during the HTTP-01 challenge process, I need to provide the generated value for the user to use on their server for domain verification.

Is it possible to return these generated values in JSON format using the lego library? I understand that these are temporary values that do not need to be stored long-term, but I would like to return them to the user during the request process for further use.

I would appreciate any clarification.

**Reply by ldez** [reply, marked as answer]
You are mixing things that a completely different: each challenge has its constraints.

For the DNS challenge, you don't need to store anything because it only relies on DNS.
For the HTTP challenge, you can do the same thing as the `HTTPProvider` implementation https://github.com/go-acme/lego/blob/master/providers/http/webroot/webroot.go

lego is integrated into several software like Traefik, 1Panel, etc. that are already doing what you are trying to do and they don't need to store anything else than the account information, private key, and certificates.

> Is it possible to return these generated values in JSON format using the lego library? 

This question doesn't make any sense to me as it depends on your code.

**Reply by disbeliefff** [reply]
Okay, thank you very much.
